torch/utils/_sympy/singleton_int.py

```diff
@@ -0,0 +1,86 @@
+import sympy
+from sympy.multipledispatch import dispatch
+
+__all__ = ["SingletonInt"]
+
+
+class SingletonInt(sympy.AtomicExpr):
+    # This is probably not super important unless we are in multiple dispatch
+    # situations with other more exotic Expr types.
+    _op_priority = 99999
+
+    # The semantics of this class should match that of SingletonSymNodeImpl in
+    # c10/core/SingletonSymNodeImpl.h
+    def __init__(self, val, coeff):
+        self._val = val
+        self._coeff = coeff
+        super().__init__()
+
+    # See NOTE [ Inequalities with SingletonInt ]
+    def _eval_Eq(self, other):
+        if (
+            isinstance(other, SingletonInt)
+            and other._val == self._val
+            and self._coeff == other._coeff
+        ):
+            return sympy.true
+        else:
+            return sympy.false
+
+    # This is necessary so that calling expr.free_symbols on exprs that contain
+    # this Singleton does not error
+    @property
+    def free_symbols(self):
+        return set()
+
+    def __mul__(self, other):
+        if isinstance(other, SingletonInt):
+            raise NotImplementedError("NYI")
```

Comment:
It's not just NYI, it doesn't semantically make sense, no?

Reply:
Good point, it seems unlikely that we'll end up implementing this.